packages/app/global-state/progressStore/progressStore.ts

```diff
@@ -0,0 +1,66 @@
+import { atom } from 'jotai';
+import { produce } from 'immer';
+import { useReducerAtom } from 'jotai/utils';
+import * as Types from './types';
+
+import EventEmitter from 'events';
```

Comment:
just curios why do we need these events and emitter logic in here ?

Reply:
With jotai we can't dispatch actions for reducer atoms outside of the component, so I ended up creating this logic so we can do it, for the progress bar state. It seems we don't use the progress bar now, but it is there, so I think it's better to keep its support.